### PR TITLE
MGDAPI-234 Add priorityClass and configure RHSSO, USERSSO and threescale to use the priorityClass

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
Add priorityClass and configure RHSSO, USERSSO and threescale to use the priorityClass

Verification steps. 
1. Install managed-api type. 
2. Confirm a new priorityclass CR called `managed-service-priority`
3. Confirm managed-api CR has the field `priorityClassName` and it's value is `managed-service-priority`
4. Confirm RHSSO and USERSSO `keycloak` statefulsets have the field `priorityClassName` with the value of `managed-service-priority` 
5. Confirm threescale deployment configs have the field `priorityClassName` with the value of `managed-service-priority` 

Extra verification step if time allows, Install managed type and confirm the above doesn't exist.

### CODE REVIEW

File: pkg/controller/installation/installation_controller.go [hunk 1]
```diff
@@ -154,13 +156,18 @@ func createInstallationCR(ctx context.Context, serverClient k8sclient.Client) er
...
 
 		logrus.Infof("Creating a %s rhmi CR with USC %s, as no CR rhmis were found in %s namespace", installType, useClusterStorage, namespace)
 
 		if installType == "" {
 			installType = string(integreatlyv1alpha1.InstallationTypeManaged)
 		}
 
+		if installType == string(integreatlyv1alpha1.InstallationTypeManagedApi) && priorityClassName == "" {
+			priorityClassName = managedServicePriorityClassName
+		}
```
Comment (sergioifg94) [inline pkg/controller/installation/installation_controller.go:169]: I see this block is reached only when the CR does not exist. When upgrading the operator from a previous version, as the installation is already complete and the CR exists, the `.spec.priorityClassName` field won't be set automatically, so the PriorityClass won't be created unless the field is set manually. If it's something that we want to support maybe we could move this logic to the Reconcile loop

**Review by sergioifg94 — COMMENTED** [review]
Verified it on a cluster, just that one comment

**Comment by pmccarthy** [comment]
@CathalOConnorRH looks like this needs rebasing

**Comment by philbrookes** [comment]
Verified locally.

**Pull request merged**